include/nbl/builtin/hlsl/cpp_compat.hlsl

```diff
@@ -38,6 +45,60 @@ namespace nbl
 namespace hlsl
 {
 
+namespace impl
+{
+    template<class T> float16_t asfloat16_t(T val) { return asfloat16(val); }
+    template<class T> float32_t asfloat32_t(T val) { return asfloat(val); }
+    template<class T> int16_t   asint16_t(T val) { return asint16(val); }
+    template<class T> int32_t   asint32_t(T val) { return asint(val); }
+    template<class T> uint16_t  asuint16_t(T val) { return asuint16(val); }
+    template<class T> uint32_t  asuint32_t(T val) { return asuint(val); }
```

Comment:
you don't need to make a named and explicitly sized `asuint`, `asint` or `asfloat`, because a X-bit float can only convert to an X-bit signed/unsigned integer

Reply:
I dont entirely understand what you suggest

Reply:
why on earth would you allow any `T` to enter these functions? constrain a little bit